VirtoCommerce.OrderModule.Web/Controllers/Api/WorkflowController.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Web.Http;
+using System.Web.Http.Description;
+using System.Web.Http.Results;
+using Newtonsoft.Json;
+using VirtoCommerce.OrderModule.Core.Models;
+using VirtoCommerce.OrderModule.Core.Services;
+using VirtoCommerce.OrderModule.Web.Security;
+using VirtoCommerce.Platform.Core.Assets;
+using VirtoCommerce.Platform.Core.Web.Security;
+
+namespace VirtoCommerce.OrderModule.Web.Controllers.Api
+{
+    [RoutePrefix("api/workflows")]
+    [CheckPermission(Permission = WorkflowPredefinedPermissions.Read)]
+    public class WorkflowController : ApiController
+    {
+        private readonly IWorkflowService _workflowService;
+        private readonly IBlobStorageProvider _blobStorageProvider;
+
+        public WorkflowController(IWorkflowService workflowService, IBlobStorageProvider blobStorageProvider)
+        {
+            _workflowService = workflowService;
+            _blobStorageProvider = blobStorageProvider;
+        }
+
+        [HttpGet]
+        [Route("{organizationId}")]
+        [ResponseType(typeof(Workflow))]
+        public IHttpActionResult Get(string organizationId)
+        {
+            var workflow = _workflowService.GetByOrganizationId(organizationId);
+            return Ok(new { data = workflow });
+        }
+
+        [HttpPost]
+        [Route("")]
+        [ResponseType(typeof(Workflow))]
+        [CheckPermission(Permission = WorkflowPredefinedPermissions.Upload)]
+        public IHttpActionResult Upload([FromBody] Workflow model)
+        {
+            if (model == null)
+                return BadRequest();
```

Comment:
throw new InvalidOperationException
with message

Reply:
it's fixed